stdlib/Test/src/Test.jl

```diff
@@ -839,6 +843,66 @@ macro test_warn(msg, expr)
     end
 end
 
+"""
+    @test_output msg expr
```

Comment:
Perhaps just define `@test_stream`? `@test_stream stdout msg expr` is IMO clearer compared to `@test_output msg expr`. Can also deprecate `@test_warn` in favor of `@test_stream stderr` then.

Reply:
I think that's a question of style. Personally I'd prefer not to have to stick a stream name in my test statement, and would avoid (unnecessarily?) deprecating an existing macro.

But if others see this differently, I am of course happy to change things :slightly_smiling_face: